### PR TITLE
Use tags and constants to build components of static pages #5782

### LINKED ISSUE #5782: Use tags and constants to build components of static pages
After porting to JSP, the static pages can pull content from tags and `Const.java` and `Config.java`, which suppresses the need for unnecessary hard-coding (thus repetition).

### PR DESCRIPTION
**Pull request by wkurniawan07**
Fixes #5782 

1. Added constant names in `Const.ViewURIs` to mimic the same function used by the error pages.
2. Removed all the hardcoded `teammates@comp.nus.edu.sg` and changed to reference to `/contact.jsp` page (or, if you would, `Const.ViewURIs.CONTACT`) to make it consistent.
3. Common tags are made for 3 purposes:
  - Overview bytes in index page
  - Features bytes in features page
  - Name list in about page; this includes all positions

Again, this would be tested best by examining the page directly: here's a [staging server](http://5-81sp-dot-teammates-wkurniawan07.appspot.com) to confirm that things are up and running.

### CODE REVIEW

File: src/main/webapp/about.jsp [hunk 1]
```diff
@@ -273,480 +189,147 @@
...
+                <ts:contributor name="Hou Guochen" />
+                <ts:contributor name="Huang Guanyang Eugene" />
+                <ts:contributor name="Huynh Van Quang" />
+                <ts:contributor name="Jiang Sheng" ghUsername="Gisonrg" />
+                <ts:contributor name="Jiang Yaoxuan" />
+                <ts:contributor name="John Kevin Tjahjadi" ghUsername="jkt001" />
+                <ts:contributor name="Karan Kamath" />
+                <ts:contributor name="Karandeep Singh Bhatia" ghUsername="karan173" />
+                <ts:contributor name="Kenneth Ho Chee Chong" ghUsername="kennho" />
+                <ts:contributor name="Khoo Yong Jie" ghUsername="YongJieYongJie" />
```
Comment (chowyb) [inline src/main/webapp/about.jsp]: Well I guess we finally get our surname in there :+1:

File: src/main/webapp/instructorHelp.jsp [hunk 2]
```diff
@@ -367,11 +370,8 @@
             <li>
                 <span class="text-bold">When you need help</span>
                 <div class="helpSectionContent">
-                    If you have a doubt or need our help, just
-                    <a href="mailto:teammates@comp.nus.edu.sg">email us
-                    </a>. We respond within 24 hours.
-                    <br>
-                    <br>
+                    If you have a doubt or need our help, just <a href="${contactPage}">contact us us</a>. We respond within 24 hours.
```
Comment (chowyb) [inline src/main/webapp/instructorHelp.jsp]: Remove the extra "us" from the `<a>` tag

File: src/main/webapp/instructorHelp.jsp [hunk 3]
```diff
@@ -458,8 +458,8 @@
                         </a>
                     </b>
                     <div class="helpSectionContent">
-                        The most likely reason for this is that the student has changed the primary email address associated with his/her Google ID. Please ask the student to email
-                        <a href="mailto:teammates@comp.nus.edu.sg">teammates@comp.nus.edu.sg</a> so that we help him to rectify the problem.
+                        The most likely reason for this is that the student has changed the primary email address associated with his/her Google ID.
+                        Please ask the student <a href="${contactPage}">contact us</a> so that we help him to rectify the problem.
```
Comment (chowyb) [inline src/main/webapp/instructorHelp.jsp]: Add an extra "to" ie. "Please ask the student to contact us"

File: src/main/java/teammates/common/util/Const.java [hunk 4]
```diff
@@ -1078,7 +1087,7 @@ public static final String getjQueryUiFilePath(boolean isDevServer) {
                 + "If you don't remember which Google account you used previously, "
                 + "email us from the same email account to which you receive TEAMMATES emails."
                 + "<br>2. You changed the primary email from a non-Gmail address to a Gmail address recently. "
-                + "In that case, <a href=\"/contact.jsp\">email us</a> "
+                + "In that case, <a href=\"" + ViewURIs.CONTACT + "\">email us</a> "
```
Comment (chowyb) [inline src/main/java/teammates/common/util/Const.java]: Maybe we could standardise this to "contact us" at the same time

File: src/main/webapp/contact.jsp [hunk 5]
```diff
@@ -1,12 +1,15 @@
...
+<c:set var="supportEmail" value="<%= Config.SUPPORT_EMAIL %>" />
 <t:staticPage currentPage="contact">
     <h1 id="caption">
         Contact Us
     </h1>
     <img src="images/contact.png" width="125px" style="padding-left: 100px;">
     <div id="contentHolder">
         <p>
-            <span class="bold">Email: </span> You can contact us at the following email address - <a href="mailto:teammates@comp.nus.edu.sg">teammates@comp.nus.edu.sg</a>
+            <span class="bold">Email: </span> You can contact us at the following email address - <a href="mailto:${supportEmail}">${supportEmail}</a>
```
Comment (chowyb) [inline src/main/webapp/contact.jsp]: Not sure if live server needs configuration due to this, but I'm leaving a note here just in case

**Comment by chowyb** [comment]
Just a few changes suggested :+1:

Wanted to add more links to the other "contact us" phrases in `instructorHelp.jsp` as well but that's a little outside the scope of this PR haha

File: src/main/webapp/contact.jsp (hunk 5 above)
Comment (wkurniawan07) [inline src/main/webapp/contact.jsp]: Nothing new here, it's just taking the value from `Config.SUPPORT_EMAIL`.

**Comment by chowyb** [comment]
Alright good to go :+1:

**Comment by damithc** [comment]
hmm... I think we should keep the static pages as simple html (sorry I didn't think of it earlier). Applying bootstrap is fine, but converting to jsp is probably not a good idea. using JSTL is definitely a no. The reason is that our static webpages are an entry point to non-javaEE developers and a good source of beginner issues. We don't want to lose that for the sake of reducing code duplication or standardization. So, any optimizations to static website should leave it accessible to non-JavaEE devs.

**Comment by wkurniawan07** [comment]
I can understand the reason against JSTL, but for JSP I don't think it's a strong reason. In fact, the reason "static webpages are an entry point to non-javaEE developers" can be looked the other way around: we can use the static webpages to introduce the most basic concepts of JSP (or even JSTL). If anything, this is beneficial since TEAMMATES is a Java web app, not a collection of static webpages.

Also, the static pages (and if I may add, the error pages) are almost 100% HTML codes with JSP component only to inject header/footer and constants from `Const.java` or `build.properties`. I really don't see the point of having to hardcode the constants for the sake of keeping the pages as HTML; it's a violation of DRY principle, and very prone to things like "forgetting to change constant ABC in page XYZ".

So I'm strongly for JSP conversion (which has been done), but do not mind converting the JSTL back to hardcoded HTMLs if it's better.

**Comment by damithc** [comment]
TEAMMATES is a java web app, but here we are talking about the product web site. As any other static website, shouldn't it be able to survive with just html, css, and js?.

**Comment by wkurniawan07** [comment]
It does and it has been so, but are there real strong reasons for keeping it pure HTML?

**Comment by damithc** [comment]
Closing this as we are not taking this path for now. Can reopen if we change the decision later.

**Pull request closed without merging**